src/v0/destinations/mp/util.test.js

```diff
@@ -98,371 +147,338 @@ describe('Mixpanel utils test', () => {
+      traits: { age: 30, address: 'kolkata' },
+      contextTraits: {},
+      setOnce: { $name: 'John', $email: 'john@example.com' },
     });
+  });
 
-    it('should add properties to the setOnce property when given setOnceProperties array with existent nested properties', () => {
-      const traits = { name: 'John', age: 30, address: { city: 'kolkata' }, isAdult: false };
-      const contextTraits = { email: 'john@example.com' };
-      const setOnceProperties = ['name', 'email', 'address.city'];
+  it('should add properties to the setOnce property when given setOnceProperties array with existent nested properties', () => {
+    const traits = { name: 'John', age: 30, address: { city: 'kolkata' }, isAdult: false };
+    const contextTraits = { email: 'john@example.com' };
+    const setOnceProperties = ['name', 'email', 'address.city'];
 
-      const result = trimTraits(traits, contextTraits, setOnceProperties);
+    const result = trimTraits(traits, contextTraits, setOnceProperties);
 
-      expect(result).toEqual({
-        traits: { age: 30, address: {}, isAdult: false },
-        contextTraits: {},
-        setOnce: { $name: 'John', $email: 'john@example.com', $city: 'kolkata' },
-      });
+    expect(result).toEqual({
+      traits: { age: 30, address: {}, isAdult: false },
+      contextTraits: {},
+      setOnce: { $name: 'John', $email: 'john@example.com', $city: 'kolkata' },
     });
   });
 });
+
+describe('generatePageOrScreenCustomEventName', () => {
+  it('should generate a custom event name when userDefinedEventTemplate contains handlebars and message object is provided', () => {
+    const message = { name: 'Home' };
+    const userDefinedEventTemplate = 'Viewed a {{ name }} page';
+    const expected = 'Viewed a Home page';
+    const result = generatePageOrScreenCustomEventName(message, userDefinedEventTemplate);
+    expect(result).toBe(expected);
+  });
+
+  it('should throw a ConfigurationError when userDefinedEventTemplate is not provided', () => {
+    const message = { name: 'Home' };
+    const userDefinedEventTemplate = undefined;
+    expect(() => {
+      generatePageOrScreenCustomEventName(message, userDefinedEventTemplate);
+    }).toThrow(ConfigurationError);
+  });
+
+  it('should return the userDefinedEventTemplate when it does not contain handlebars', () => {
+    const message = { name: 'Home' };
+    const userDefinedEventTemplate = 'Viewed a Home page';
+    const expected = 'Viewed a Home page';
+    const result = generatePageOrScreenCustomEventName(message, userDefinedEventTemplate);
+    expect(result).toBe(expected);
+  });
+
+  it('should return a event name when message object is not provided', () => {
+    const message = undefined;
+    const userDefinedEventTemplate = 'Viewed a {{ path }} page';
+    const expected = 'Viewed a  page';
+    const result = generatePageOrScreenCustomEventName(message, userDefinedEventTemplate);
+    expect(result).toBe(expected);
+  });
```

Comment:
What will happen if multiple handlebers are there in template? Like if the template is `{{ name }} page viewed from {{ context.page.url.host }}`

Reply:
Added the testcase for it.